### DISCUSSION TITLE
Timeouts for CheckPermission and other functions?

### DISCUSSION THREAD

**Discussion by sofiapag** [authzed/spicedb#2047] [Q&A]
Hi there, is there a default timeout for CheckPermission and other spiceDB client functions? If not, what is the recommendation for how to set a timeout on these calls?

Thanks!

**Comment by vroldanbet** [comment]
Hey @sofiapag 👋🏻  It depends on what client you are using, but yes [gRPC supports the concept of deadline](https://grpc.io/docs/guides/deadlines/), which is effectively a timeout and can be set on the client side. How this is done in practice depends on each language, but for example, in Go this is achieved by [setting a deadline for the context](https://pkg.go.dev/context#WithTimeout)

**Comment by tstirrat15** [comment]
Note that the various SpiceDB clients do not set deadlines by default.
